ksqldb-functional-tests/src/test/resources/query-validation-tests/row-offset-and-partition.json

```diff
@@ -119,6 +150,622 @@
+        "CREATE STREAM b (id INT KEY, foo STRING) WITH (kafka_topic='test_topic_2', value_format='JSON');",
+        "CREATE STREAM output AS SELECT a.ROWPARTITION AS rpa, a.ROWOFFSET AS roa, b.ROWPARTITION as rpb, b.ROWOFFSET as rob FROM a JOIN b ON a.ROWPARTITION = b.ROWPARTITION;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp":  7},
+        {"topic":  "test_topic_2", "key": 4, "value": {"foo": "d"}, "timestamp":  8},
+        {"topic":  "test_topic_2", "key": 5, "value": {"foo": "e"}, "timestamp":  9},
+        {"topic":  "test_topic_2", "key": 6, "value": {"foo": "f"}, "timestamp":  10}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"RP": 0, "RO": 0, "RT": 5}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 2, "value": {"RP": 0, "RO": 1, "RT": 6}, "timestamp": 6},
+        {"topic":  "OUTPUT", "key": 3, "value": {"RP": 0, "RO": 2, "RT": 7}, "timestamp": 7}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, RP INT, RO BIGINT, RT BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "join on ROWOFFSET - INCOMPLETE",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM a (id INT KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM b (id INT KEY, bar STRING) WITH (kafka_topic='test_topic_2', value_format='JSON');",
+        "CREATE STREAM output AS SELECT foo, bar, a.ROWOFFSET AS roa, b.ROWOFFSET as rob FROM a JOIN b  WITHIN 5 SECONDS ON a.ROWOFFSET = b.ROWOFFSET;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp":  7},
+        {"topic":  "test_topic_2", "key": 4, "value": {"bar": "d"}, "timestamp":  8},
+        {"topic":  "test_topic_2", "key": 5, "value": {"bar": "e"}, "timestamp":  9},
+        {"topic":  "test_topic_2", "key": 6, "value": {"bar": "f"}, "timestamp":  10}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 0, "value": {"foo": "a", "bar": "d","roa": 0, "rob": 0}, "timestamp": 8},
+        {"topic":  "OUTPUT", "key": 0, "value": {"foo":  "b", "bar": "e", "roa": 1, "rob": 1}, "timestamp": 9},
+        {"topic":  "OUTPUT", "key": 0, "value": {"foo":  "c", "bar": "f","roa": 2, "rob": 2}, "timestamp": 10}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "FOO STRING, BAR STRING, ROA BIGINT, ROB BIGINT"}
```

Comment:
This test fails because ROA is not in the projection, but ROB is. On manual testing, both are in the projection. Any ideas what this might be?

Reply:
Is ROA in the key schema instead of the value schema? (Just throwing out a guess.)

Reply:
That was indeed the problem. Is this expected behavior? Seems a little strange.

Reply:
It's expected. ROA is the join key.